crates/sail-plan/src/data_source/mod.rs

```
@@ -0,0 +1,57 @@
+use std::collections::HashMap;
+
+use serde::Deserialize;
+
+use crate::error::{PlanError, PlanResult};
+
+pub mod csv;
+
+const fn default_true() -> bool {
+    true
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct ConfigItem {
+    key: String,
+    #[serde(default)]
+    alias: Vec<String>,
+    default: String,
+    #[expect(unused)]
+    description: String,
+    #[serde(default = "default_true")]
+    supported: bool,
+}
+
+impl ConfigItem {
+    // user_options_normalized is expected to be a HashMap with all keys in lowercase
+    fn resolve_value(&self, user_options_normalized: &HashMap<String, String>) -> String {
+        if let Some(value) = user_options_normalized.get(&self.key.to_lowercase()) {
+            return value.clone();
+        }
+        for alias in &self.alias {
```

If both the key and the alias are present in the map, such duplication is silently ignored and the user does not know which one is picked without looking at the code. Let's add a TODO here for such situation.